src/common/util/include/openvino/util/file_util.hpp

```diff
@@ -12,6 +12,20 @@
 
 #include "openvino/util/util.hpp"
 
+#if defined(__linux__) && !defined(__ANDROID__)
+#    if (__cplusplus >= 201703L)
```

Comment:
What if build with older C++ version? Why there is not need to check symlinks with older compiler?

Reply:
Here is to make sure we are using the correct header file of filesystem
For > 2017, the header file is <filesystem>
For order version, the header file is <experimental/filesystem>
The symlink also be checked in older compiler